### PR TITLE
Gas pipes

### PR DESCRIPTION
**Pull request by Bod9001**
addes gasPipes
pumps, filters, mixers, vent, scrubbers
A few Mapping tool changes as well

*Coloured pipes need the underfloor layer material to be Changed to defaults Sprite to get the colours to work
Coloured pipes
requires mapping pipes 
the space wind might be a bit laggy tho Need optimisation and investigation

I would say basic level pipes

Square Station pipes done

### CODE REVIEW

**Comment by ThatDan123** [comment]
Whats the deal with all the deleted files, are they unused?

**Comment by ThatDan123** [comment]
looks like a lot of the test failing is sprite references

**Comment by Bod9001** [comment]
unused, tile preview stuff imo could be done in a better way and is for items that don't work

**Comment by corp-0** [comment]
3k files changed e_e

**Comment by Bod9001** [comment]
tests should be good now

**Comment by corp-0** [comment]
tests passed, how good is this for some testing in staging :eyes: ?

File: UnityProject/Assets/Scripts/Construction/SpawnOnWrench.cs [hunk 1]
```diff
@@ -0,0 +1,25 @@
+﻿using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+public class SpawnOnWrench : MonoBehaviour, ICheckedInteractable<HandApply>
+{
+	public GameObject toSpawn = null;
```
Comment (corp-0) [inline UnityProject/Assets/Scripts/Construction/SpawnOnWrench.cs]: if this is meant to be set in inspector, use a private property with ``[SerializeField]`` decorator and a public getter. Also add a ``Tooltip`` decorator. 

[Guidelines reference](https://unitystation.github.io/unitystation/contribution-guides/Development-Standards-Guide/#component-and-scriptable-object-design)

File: UnityProject/Assets/Scripts/FluidPipe/Delta.cs [hunk 2]
```diff
@@ -0,0 +1,18 @@
+﻿using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+public class Delta : MonoBehaviour
```
Comment (corp-0) [inline UnityProject/Assets/Scripts/FluidPipe/Delta.cs]: what does this class do?

File: UnityProject/Assets/Scripts/FluidPipe/Filter.cs [hunk 3]
```diff
@@ -0,0 +1,143 @@
+﻿using System.Collections;
+using System.Collections.Generic;
+using Atmospherics;
+using UnityEngine;
+
+namespace Pipes
+{
+	public class Filter : MonoPipe
+	{
+		public SpriteHandler spriteHandlerOverlay = null;
```
Comment (corp-0) [inline UnityProject/Assets/Scripts/FluidPipe/Filter.cs:10]: Same thing. Make it private with a public getter and  [SerializeField] decorator.

**Review by corp-0 — COMMENTED** [review]
I did what I could, my browser dies when trying to diff so many files xD.

Make sure to use private fields like said and also if something is not meant to be set in inspector, use a public getter/setter or [HideInInspector] decorator. Let's keep our componentes tidy and easy to understand at a glance!

File: UnityProject/Assets/Scripts/FluidPipe/Delta.cs (hunk 2 above)
Comment (Bod9001) [reply UnityProject/Assets/Scripts/FluidPipe/Delta.cs]: accidentally added

**Comment by Bod9001** [comment]
thanks for review, will do changes tomorrow

**Comment by ThatDan123** [comment]
Bod does your PR still support vents and scrubbers working without pipes? This might be needed for awaysites, also the maps if they dont get converted before next update.

~Also i noticed in your PR that theres no scripts on the vents/scrubbers is this intended?~

~As in the future the ability to turn them on/off and adjust pressures will be needed, this also links to the above question about having the work as they do currently by having a toggle on them.~

EDIT: Oh i see youve made a new vent/scrubber prefab, but the functionality of magically creating air would be nice to have still

**Comment by ThatDan123** [comment]
Getting this when attempting to wrench down a vent.

EDIT: happens on the old vent prefab, so nvm

Also all mapped vents can be picked up.

`NullReferenceException: Object reference not set to an instance of an object
InteractablePipe.ServerPerformInteraction (PositionalHandApply interaction) (at Assets/Scripts/Tilemaps/Behaviours/Meta/Atmospherics/Pipenets/InteractablePipe.cs:28)
RequestInteractMessage.ServerCheckAndTrigger[T] (T interaction, System.Collections.Generic.IEnumerable`1[T] interactables) (at Assets/Scripts/Messages/Client/Interaction/RequestInteractMessage.cs:347)
RequestInteractMessage.ProcessInteraction[T] (T interaction, UnityEngine.GameObject processorObj) (at Assets/Scripts/Messages/Client/Interaction/RequestInteractMessage.cs:329)
RequestInteractMessage.Process () (at Assets/Scripts/Messages/Client/Interaction/RequestInteractMessage.cs:129)
GameMessageBase.Process (Mirror.NetworkConnection sentBy) (at Assets/Scripts/Messages/GameMessageBase.cs:25)
ClientMessage.Process (Mirror.NetworkConnection sentBy) (at Assets/Scripts/Messages/Client/ClientMessage.cs:14)
GameMessageBase.PreProcess (Mirror.NetworkConnection sentBy, GameMessageBase b) (at Assets/Scripts/Messages/GameMessageBase.cs:18)
Mirror.MessagePacker+<>c__DisplayClass6_0`2[T,C].<MessageHandler>b__0 (Mirror.NetworkConnection conn, Mirror.NetworkReader reader, System.Int32 channelId) (at Assets/Mirror/Runtime/MessagePacker.cs:146)
Mirror.NetworkConnection.InvokeHandler (System.Int32 msgType, Mirror.NetworkReader reader, System.Int32 channelId) (at Assets/Mirror/Runtime/NetworkConnection.cs:219)
Mirror.NetworkConnection.TransportReceive (System.ArraySegment`1[T] buffer, System.Int32 channelId) (at Assets/Mirror/Runtime/NetworkConnection.cs:272)
Mirror.ULocalConnectionToServer.Send (System.ArraySegment`1[T] segment, System.Int32 channelId) (at Assets/Mirror/Runtime/LocalConnections.cs:97)
Mirror.NetworkConnection.Send[T] (T msg, System.Int32 channelId) (at Assets/Mirror/Runtime/NetworkConnection.cs:150)
Mirror.NetworkClient.Send[T] (T message, System.Int32 channelId) (at Assets/Mirror/Runtime/NetworkClient.cs:248)
ClientMessage.Send () (at Assets/Scripts/Messages/Client/ClientMessage.cs:19)
RequestInteractMessage.Send[T] (T interaction, IBaseInteractable`1[T] interactableComponent) (at Assets/Scripts/Messages/Client/Interaction/RequestInteractMessage.cs:474)
InteractionUtils.RequestInteract[T] (T interaction, IBaseInteractable`1[T] interactableComponent) (at Assets/Scripts/Input System/InteractionV2/InteractionUtils.cs:31)
InteractionUtils.ClientCheckAndTrigger[T] (IBaseInteractable`1[T] interactable, T interaction) (at Assets/Scripts/Input System/InteractionV2/InteractionUtils.cs:53)
MouseInputController.CheckHandApply (UnityEngine.GameObject target) (at Assets/Scripts/Input System/MouseInputController.cs:392)
MouseInputController.CheckClick () (at Assets/Scripts/Input System/MouseInputController.cs:334)
MouseInputController.CheckClickInteractions (System.Boolean includeAimApply) (at Assets/Scripts/Input System/MouseInputController.cs:259)
MouseInputController.CheckMouseInput () (at Assets/Scripts/Input System/MouseInputController.cs:172)
MouseInputController.LateUpdate () (at Assets/Scripts/Input System/MouseInputController.cs:99)`

**Comment by Bod9001** [comment]
is it all good? or should I wait for the maps to be finished

**Comment by ThatDan123** [comment]
Theres still issues with the filter UI, test on square and pumps start with the on Sprite even though theyre off

**Comment by ThatDan123** [comment]
Also can you add the reflected versions of the filter and mixer

image

**Comment by Bod9001** [comment]
done

**Comment by corp-0** [comment]
all done and said, I'm merging

**Comment by waffielz** [comment]
jeez, 3k

**Pull request merged**